src/firstboot/firstboot.c

```diff
@@ -803,6 +804,27 @@ static int process_root_password(void) {
         return 0;
 }
 
+static int process_kernel_cmdline(void) {
+        const char *etc_kernel_cmdline;
+        int r;
+
+        etc_kernel_cmdline = prefix_roota(arg_root, "/etc/kernel/cmdline");
+        if (laccess(etc_kernel_cmdline, F_OK) >= 0 && !arg_force)
+                return 0;
+
+        if (isempty(arg_kernel_cmdline))
```

Comment:
if specified empty, and arg_force is set we should probably unlink the file, no? (and rmdir /etc/kernel/ if that works afterwards)

Reply:
If /etc/kernel/cmdline doesn't exist, kernel-install uses /proc/cmdline so there would be a difference in behaviour. Maybe have not specified mean /proc/cmdline and empty mean /etc/kernel/cmdline file?

Reply:
hmm, if the parameter isn't specified i wouldn't make any changes. But you are right, it might make sense to set it to the empty string. But your current code treats option not specified and value set to the empty string the same way, after all you use isempty() here, which does exactly that check.

If figure this line should just be changed to `if (!arg_kernel_cmdline)`